spi/src/main/java/org/opensearch/security/spi/resources/client/NoopResourceSharingClient.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.security.spi.resources.client;
+
+import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.OpenSearchStatusException;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.security.spi.resources.ShareableResource;
+import org.opensearch.security.spi.resources.sharing.ResourceSharing;
+import org.opensearch.security.spi.resources.sharing.SharedWithActionGroup;
+
+/**
+ * Creates a noop client that will be used when security is disabled,
+ *
+ * @opensearch.experimental
+ */
+public final class NoopResourceSharingClient implements ResourceSharingClient {
+
+    private static final Logger log = LogManager.getLogger(NoopResourceSharingClient.class);
+
+    /**
+     * Constructs a NoopResourceSharingClient
+     */
+    public NoopResourceSharingClient() {}
+
+    /**
+     * Returns true as a no-op implementation.
+     *
+     * @param resourceId    The ID of the resource to verify.
+     * @param resourceIndex The index in which the resource resides.
+     * @param listener      Callback that receives {@code true} if access is granted, {@code false} otherwise.
+     */
+    @Override
+    public void verifyResourceAccess(String resourceId, String resourceIndex, ActionListener<Boolean> listener) {
+        String message = getFeatureDisabledReason() + " " + "Access to resource is automatically granted.";
+        log.debug("{}", message);
+        listener.onResponse(true);
+    }
+
+    /**
+     * Throws 501 exception as a no-op implementation.
+     *
+     * @param resourceId    The ID of the resource to share.
+     * @param resourceIndex The index containing the resource.
+     * @param recipients    The recipients of the resource, including users, roles, and backend roles.
+     * @param listener      Callback receiving the updated {@link ResourceSharing} document.
+     */
+    @Override
+    public void shareResource(
```

Comment:
Can we shorten these method names to names like `share` or `revoke`. This is already a ResourceSharingClient so having resource in the method names is overly verbose.

When is `verifyResourceAccess` used?

Reply:
> Can we shorten these method names to names like share or revoke. This is already a ResourceSharingClient so having resource in the method names is overly verbose.

Sure.

> When is verifyResourceAccess used?

Here is a usage in [ADPlugin](https://github.com/opensearch-project/anomaly-detection/pull/1400/files#diff-cf370d40fdd2abc404283fa1e37768646f30ee1b8bfbb0b5c5302fc8a2a080fcR721). Another one in [ml-plugin](https://github.com/opensearch-project/ml-commons/pull/3715/files#diff-7e30388762cbeaaf7af2391b1f6a623882537b3363ba3d3801f62223b98097aaR173).